### DISCUSSION TITLE
Dynamic Sandbox for Listing Products

### DISCUSSION THREAD

**Discussion by janhellopine** [amzn/selling-partner-api-models#3939] [General] [answered]
Is it true that there is no dynamic sandbox for the Listings Items API? 
Is there a best practice in place to test the creation of products?

**Comment by rugved1991** [comment, marked as answer]
Hi @janhellopine,

We launched today a new change in the Listings API to check for validation errors while creating a listing on Amazon. Please review this announcement for more information about the validation preview:
https://developer-docs.amazon.com/sp-api/changelog/update-listings-items-api-v2021-08-01-now-supports-previewing-errors

Best,
Rugved
Solutions Architect, SP-API

**Comment by janhellopine** [comment]
That's a game changer, thank you!

**Comment by mdziew** [comment]
I am not sure if I can still ask here. This is indeed a nice feature. It works for me for `putListingsItem`, but unfortunately not for `patchListingsItem`. For the latter I get 429 and this response every time:
```json
{
    "errors": [
        {
            "code": "QuotaExceeded",
            "message": "You exceeded your quota for the requested resource.",
            "details": ""
        }
    ]
}
```

My sample request is:
```json
{
    "productType": "ACCESSORY_OR_PART_OR_SUPPLY",
    "patches": [
        {
            "op": "replace",
            "path": "/attributes/item_name",
            "value": [
                {
                    "language_tag": "es_ES",
                    "marketplace_id": "A1RKKUPIHCS9HS",
                    "value": "some name"
                }
            ]
        }
    ]
}
```

NOTE: It happens during manual tests from postman so I don't think I am exceeding limits.
